bindings/virtio/pagetable.S

```diff
@@ -2,11 +2,529 @@
+	.quad 0x182000 + 0x3
+	.quad 0x183000 + 0x3
+	.quad 0x184000 + 0x3
+	.quad 0x185000 + 0x3
+	.quad 0x186000 + 0x3
+	.quad 0x187000 + 0x3
+	.quad 0x188000 + 0x3
+	.quad 0x189000 + 0x3
+	.quad 0x18a000 + 0x3
+	.quad 0x18b000 + 0x3
+	.quad 0x18c000 + 0x3
+	.quad 0x18d000 + 0x3
+	.quad 0x18e000 + 0x3
+	.quad 0x18f000 + 0x3
+	.quad 0x190000 + 0x3
+	.quad 0x191000 + 0x3
+	.quad 0x192000 + 0x3
+	.quad 0x193000 + 0x3
+	.quad 0x194000 + 0x3
+	.quad 0x195000 + 0x3
+	.quad 0x196000 + 0x3
+	.quad 0x197000 + 0x3
+	.quad 0x198000 + 0x3
+	.quad 0x199000 + 0x3
+	.quad 0x19a000 + 0x3
+	.quad 0x19b000 + 0x3
+	.quad 0x19c000 + 0x3
+	.quad 0x19d000 + 0x3
+	.quad 0x19e000 + 0x3
+	.quad 0x19f000 + 0x3
+	.quad 0x1a0000 + 0x3
+	.quad 0x1a1000 + 0x3
+	.quad 0x1a2000 + 0x3
+	.quad 0x1a3000 + 0x3
+	.quad 0x1a4000 + 0x3
+	.quad 0x1a5000 + 0x3
+	.quad 0x1a6000 + 0x3
+	.quad 0x1a7000 + 0x3
+	.quad 0x1a8000 + 0x3
+	.quad 0x1a9000 + 0x3
+	.quad 0x1aa000 + 0x3
+	.quad 0x1ab000 + 0x3
+	.quad 0x1ac000 + 0x3
+	.quad 0x1ad000 + 0x3
+	.quad 0x1ae000 + 0x3
+	.quad 0x1af000 + 0x3
+	.quad 0x1b0000 + 0x3
+	.quad 0x1b1000 + 0x3
+	.quad 0x1b2000 + 0x3
+	.quad 0x1b3000 + 0x3
+	.quad 0x1b4000 + 0x3
+	.quad 0x1b5000 + 0x3
+	.quad 0x1b6000 + 0x3
+	.quad 0x1b7000 + 0x3
+	.quad 0x1b8000 + 0x3
+	.quad 0x1b9000 + 0x3
+	.quad 0x1ba000 + 0x3
+	.quad 0x1bb000 + 0x3
+	.quad 0x1bc000 + 0x3
+	.quad 0x1bd000 + 0x3
+	.quad 0x1be000 + 0x3
+	.quad 0x1bf000 + 0x3
+	.quad 0x1c0000 + 0x3
+	.quad 0x1c1000 + 0x3
+	.quad 0x1c2000 + 0x3
+	.quad 0x1c3000 + 0x3
+	.quad 0x1c4000 + 0x3
+	.quad 0x1c5000 + 0x3
+	.quad 0x1c6000 + 0x3
+	.quad 0x1c7000 + 0x3
+	.quad 0x1c8000 + 0x3
+	.quad 0x1c9000 + 0x3
+	.quad 0x1ca000 + 0x3
+	.quad 0x1cb000 + 0x3
+	.quad 0x1cc000 + 0x3
+	.quad 0x1cd000 + 0x3
+	.quad 0x1ce000 + 0x3
+	.quad 0x1cf000 + 0x3
+	.quad 0x1d0000 + 0x3
+	.quad 0x1d1000 + 0x3
+	.quad 0x1d2000 + 0x3
+	.quad 0x1d3000 + 0x3
+	.quad 0x1d4000 + 0x3
+	.quad 0x1d5000 + 0x3
+	.quad 0x1d6000 + 0x3
+	.quad 0x1d7000 + 0x3
+	.quad 0x1d8000 + 0x3
+	.quad 0x1d9000 + 0x3
+	.quad 0x1da000 + 0x3
+	.quad 0x1db000 + 0x3
+	.quad 0x1dc000 + 0x3
+	.quad 0x1dd000 + 0x3
+	.quad 0x1de000 + 0x3
+	.quad 0x1df000 + 0x3
+	.quad 0x1e0000 + 0x3
+	.quad 0x1e1000 + 0x3
+	.quad 0x1e2000 + 0x3
+	.quad 0x1e3000 + 0x3
+	.quad 0x1e4000 + 0x3
+	.quad 0x1e5000 + 0x3
+	.quad 0x1e6000 + 0x3
+	.quad 0x1e7000 + 0x3
+	.quad 0x1e8000 + 0x3
+	.quad 0x1e9000 + 0x3
+	.quad 0x1ea000 + 0x3
+	.quad 0x1eb000 + 0x3
+	.quad 0x1ec000 + 0x3
+	.quad 0x1ed000 + 0x3
+	.quad 0x1ee000 + 0x3
+	.quad 0x1ef000 + 0x3
+	.quad 0x1f0000 + 0x3
+	.quad 0x1f1000 + 0x3
+	.quad 0x1f2000 + 0x3
+	.quad 0x1f3000 + 0x3
+	.quad 0x1f4000 + 0x3
+	.quad 0x1f5000 + 0x3
+	.quad 0x1f6000 + 0x3
+	.quad 0x1f7000 + 0x3
+	.quad 0x1f8000 + 0x3
+	.quad 0x1f9000 + 0x3
+	.quad 0x1fa000 + 0x3
+	.quad 0x1fb000 + 0x3
+	.quad 0x1fc000 + 0x3
+	.quad 0x1fd000 + 0x3
+	.quad 0x1fe000 + 0x3
+	.quad 0x1ff000 + 0x3
+
 .align 0x1000
 cpu_pd:
-	.quad 0x0000000000000000 + 0x3 + 0x80
+	.quad cpu_pt0 + 0x3
```

Comment:
Don't know what flags are the `0x80`s adding, but shouldn't they be added as well to the `pte`s?

Reply:
Definitely not, that bit is `PAT` in a 4kB PTE.

Reply:
...and in a PDE it's PS, so in cpu_pd[0] it must be zero, otherwise it'd be a 2MB page.